### DISCUSSION TITLE
Component script is not executed

### DISCUSSION THREAD

**Discussion by DMotix** [livewire/livewire#6754] [1. Help] [answered]
I have a component of a select2, which has both its html and the script that initializes it

```
@props([
  'disabled' => false,
  'placeholder' => 'Select',
  'options' => [],
  'customClass' => '',
  'selected' => ''
])
<select {{ $disabled ? 'disabled' : '' }} {!! $attributes->merge(['class' => '' . $customClass . ' select2 bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500']) !!}>
	<option value="">{{ $placeholder }}</option>
	@foreach($options as $id => $label)
	@if ($selected == $id)
		<option selected value="{{ $id }}">{{ $label }}</option>
	@else
		<option value="{{ $id }}">{{ $label }}</option>
	@endif
		
	@endforeach
</select>
@push('scripts')

<script type="module">
	console.log('INSIDE SCRIPT');
	$(document).ready(function() {
		console.log('DOCUMENT READY');
		$('.{{ $customClass }}').select2();
	});
</script>
@endpush
```

If I put it in a main page it works and executes the console.log that you see in the code, but if I add it inside a component, for example a modal component inside that same page the js of the select2 component does not execute. Does anyone understand why this happens? thanks in advance.

The component is named in the .blade files as follows:

`<x-select2 :selected=$code :customClass=$customC :placeholder="'Choose an option'" :options=$options :name=$customC/>`

Livewire: 3.0.2
Php: 8.2
Laravel: 10
Server: Local(laragon)

**Comment by PhiloNL** [comment, marked as answer]
I don't think you can use `<script>` inside a modal. Try using Alpine instead:

`<div x-data x-init="$($el).select2();"></div>`
